RedBookPlayer/PlayableDisc.cs

```diff
@@ -0,0 +1,662 @@
+        /// <returns>True if the track was changed, false otherwise</returns>
+        public bool PreviousIndex(bool changeTrack)
+        {
+            if(_image == null)
+                return false;
+
+            if(CurrentTrackIndex - 1 < _image.Tracks[CurrentTrackNumber].Indexes.Keys.Min())
+            {
+                if(changeTrack)
+                {
+                    PreviousTrack();
+                    CurrentSector = (ulong)_image.Tracks[CurrentTrackNumber].Indexes.Values.Max();
+                    return true;
+                }
+            }
+            else
+            {
+                CurrentSector = (ulong)_image.Tracks[CurrentTrackNumber].Indexes[--CurrentTrackIndex];
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Fast-forward playback by 75 sectors, if possible
+        /// </summary>
+        public void FastForward()
+        {
+            if(_image == null)
+                return;
+
+            CurrentSector = Math.Min(_image.Info.Sectors - 1, CurrentSector + 75);
+        }
+
+        /// <summary>
+        /// Rewind playback by 75 sectors, if possible
+        /// </summary>
+        public void Rewind()
+        {
+            if(_image == null)
+                return;
+
+            if(CurrentSector >= 75)
+                CurrentSector -= 75;
+        }
+
+        /// <summary>
+        /// Toggle de-emphasis processing
+        /// </summary>
+        /// <param name="enable">True to apply de-emphasis, false otherwise</param>
+        public void ToggleDeEmphasis(bool enable) => ApplyDeEmphasis = enable;
+
+        #endregion
+
+        #region Helpers
+
+        /// <summary>
+        /// Load the first valid track in the image
+        /// </summary>
+        public void LoadFirstTrack()
+        {
+            CurrentTrackNumber = 0;
+            LoadTrack(CurrentTrackNumber);
+        }
+
+        /// <summary>
+        /// Read sector data from the base image starting from the current sector
+        /// </summary>
+        /// <param name="sectorsToRead">Current number of sectors to read</param>
+        /// <returns>Byte array representing the read sectors, if possible</returns>
+        public byte[] ReadSectors(uint sectorsToRead) => _image.ReadSectors(CurrentSector, sectorsToRead);
+
+        /// <summary>
+        /// Set the total indexes from the current track
+        /// </summary>
+        public void SetTotalIndexes()
+        {
+            if(_image == null)
+                return;
+
+            TotalIndexes = _image.Tracks[CurrentTrackNumber].Indexes.Keys.Max();
+        }
+
+        /// <summary>
+        /// Generate a CDFullTOC object from the current image
+        /// </summary>
+        /// <returns>CDFullTOC object, if possible</returns>
+        /// <remarks>Copied from <see cref="Aaru.DiscImages.CloneCd"/></remarks>
+        private bool GenerateTOC()
```

Comment:
Probably should go to Aaru.Decoders, as it can be used by more than one image. No need to do so in this PR if you open a FR in Aaru.

Reply:
Opened https://github.com/aaru-dps/Aaru/issues/629